hpat/tests/test_series.py

```diff
@@ -1556,6 +1556,96 @@ def test_impl(A):
         S = pd.Series(np.random.ranf(n))
         np.testing.assert_array_equal(hpat_func(S), test_impl(S))
 
+    def test_series_idxmin_str(self):
+        def test_impl(S):
+            return S.idxmin()
+        hpat_func = hpat.jit(test_impl)
+
+        S = pd.Series([8, 6, 34, np.nan], ['a', 'ab', 'abc', 'c'])
+        print(hpat_func(S))
+        print(test_impl(S))
+        self.assertEqual(hpat_func(S), test_impl(S))
+
+    @unittest.skip("Cant return 2 types: string or nan in one case")
+    def test_series_idxmin_str_idx(self):
+        def test_impl(S):
+            return S.idxmin(skipna=False)
+
+        hpat_func = hpat.jit(test_impl)
+
+        S = pd.Series([8, 6, 34, np.nan], ['a', 'ab', 'abc', 'c'])
+        print(hpat_func(S))
+        print(test_impl(S))
+        self.assertEqual(hpat_func(S), test_impl(S))
+
+    def test_series_idxmin_no(self):
+        def test_impl(S):
+            return S.idxmin()
+        hpat_func = hpat.jit(test_impl)
+
+        S = pd.Series([8, 6, 34, np.nan])
+        self.assertEqual(hpat_func(S), test_impl(S))
+
+    @unittest.skip("Enable after fixing index")
+    def test_series_idxmin_int(self):
+        def test_impl(S):
+            return S.idxmin()
+        hpat_func = hpat.jit(test_impl)
+
+        S = pd.Series([1, 2, 3], [4, 45, 14])
+        print(hpat_func(S))
+        print(test_impl(S))
+        self.assertEqual(hpat_func(S), test_impl(S))
+
+    @unittest.skip("Need index fix")
+    def test_series_idxmin(self):
+        def test_series_idxmin_impl(S):
+            return S.idxmin()
+
+        hpat_func = hpat.jit(test_series_idxmin_impl)
+
+        test_input_data = []
+        data_simple = [[6, 6, 2, 1, 3, 3, 2, 1, 2],
+                       [1.1, 0.3, 2.1, 1, 3, 0.3, 2.1, 1.1, 2.2],
+                       [6, 6.1, 2.2, 1, 3, 3, 2.2, 1, 2],
+                       ]
+
+        data_extra = [[np.nan, np.nan, np.nan, np.nan],
```

Comment:
I think it's better to have one 'series_data' list (instead of data_simple + data_extra) with it's elements covering all possible situations (i.e. containing numbers, NaNs or both, in different combinations). There's no obvious benefit of using loops in unittests - better identify and cover different cases yourself.